mkosi/qemu.py

```diff
@@ -660,16 +624,49 @@ def run_qemu(args: Args, config: Config) -> None:
         ]
 
     # QEMU has built-in logic to look for the BIOS firmware so we don't need to do anything special for that.
-    if firmware == QemuFirmware.uefi:
-        cmdline += ["-drive", f"if=pflash,format=raw,readonly=on,file={ovmf}"]
+    if firmware.is_uefi():
+        assert ovmf
+        cmdline += ["-drive", f"if=pflash,format={ovmf.format},readonly=on,file={ovmf.firmware}"]
     notifications: dict[str, str] = {}
 
     with contextlib.ExitStack() as stack:
-        if firmware == QemuFirmware.uefi:
+        if firmware.is_uefi():
+            assert ovmf
             ovmf_vars = stack.enter_context(tempfile.NamedTemporaryFile(prefix="mkosi-ovmf-vars"))
-            shutil.copy2(config.qemu_firmware_variables or find_ovmf_vars(config), Path(ovmf_vars.name))
-            cmdline += ["-drive", f"file={ovmf_vars.name},if=pflash,format=raw"]
-            if ovmf_supports_sb:
+            ovmf_vars_format = (
+                "raw"
+                if config.qemu_firmware_variables not in (None, Path("custom"), Path("microsoft"))
+                else ovmf.vars_format
+            )
```

Comment:
As soon as the ternary isn't a oneliner anymore, it is shorted as a proper if-statement.
```suggestion
            if config.qemu_firmware_variables not in (None, Path("custom"), Path("microsoft"))
                 ovmf_vars_format = "raw"
            else:
                 ovmf_vars_format = ovmf.vars_format
```